app/assets/javascripts/discourse/app/components/d-modal-legacy.js

```diff
@@ -0,0 +1,251 @@
+      return this.modalBodyData.dismissable;
+    } else {
+      return true;
+    }
+  }
+
+  get title() {
+    if (this.modalBodyData.title) {
+      return I18n.t(this.modalBodyData.title);
+    } else if (this.modalBodyData.rawTitle) {
+      return this.modalBodyData.rawTitle;
+    } else {
+      return this.args.title;
+    }
+  }
+
+  get subtitle() {
+    if (this.modalBodyData.subtitle) {
+      return I18n.t(this.modalBodyData.subtitle);
+    }
+
+    return this.modalBodyData.rawSubtitle || this.args.subtitle;
+  }
+
+  get headerClass() {
+    return this.modalBodyData.headerClass;
+  }
+
+  get panels() {
+    return this.args.panels;
+  }
+
+  get errors() {
+    return this.args.errors;
+  }
+
+  @action
+  setupListeners(element) {
+    this.appEvents.on("modal:body-shown", this._modalBodyShown);
+    this.appEvents.on("modal-body:flash", this._flash);
+    this.appEvents.on("modal-body:clearFlash", this._clearFlash);
+    document.documentElement.addEventListener(
+      "keydown",
+      this._handleModalEvents
+    );
+    this.wrapperElement = element;
+  }
+
+  @action
+  cleanupListeners() {
+    this.appEvents.off("modal:body-shown", this._modalBodyShown);
+    this.appEvents.off("modal-body:flash", this._flash);
+    this.appEvents.off("modal-body:clearFlash", this._clearFlash);
+    document.documentElement.removeEventListener(
+      "keydown",
+      this._handleModalEvents
+    );
+  }
+
+  get ariaLabelledby() {
+    if (this.modalBodyData.titleAriaElementId) {
+      return this.modalBodyData.titleAriaElementId;
+    } else if (this.args.titleAriaElementId) {
+      return this.args.titleAriaElementId;
+    } else if (this.args.title) {
+      return "discourse-modal-title";
+    }
+  }
+
+  get modalClass() {
+    return this.modalBodyData.modalClass || this.args.modalClass;
+  }
+
+  triggerClickOnEnter(e) {
+    if (!this.submitOnEnter) {
+      return false;
+    }
+
+    // skip when in a form or a textarea element
+    if (
+      e.target.closest("form") ||
+      (document.activeElement && document.activeElement.nodeName === "TEXTAREA")
+    ) {
+      return false;
+    }
+
+    return true;
+  }
+
+  @action
+  handleMouseDown(e) {
+    if (!this.dismissable) {
+      return;
+    }
+
+    if (
+      e.target.classList.contains("modal-middle-container") ||
+      e.target.classList.contains("modal-outer-container")
+    ) {
+      // Send modal close (which bubbles to ApplicationRoute) if clicked outside.
+      // We do this because some CSS of ours seems to cover the backdrop and makes
+      // it unclickable.
+      return this.args.closeModal?.("initiatedByClickOut");
+    }
+  }
+
+  @bind
+  _modalBodyShown(data) {
+    if (this.isDestroying || this.isDestroyed) {
+      return;
+    }
+
+    if (data.fixed) {
+      this.modal.hidden = false;
+    }
+
+    this.modalBodyData = data;
+
+    next(() => {
+      schedule("afterRender", () => {
+        this._trapTab();
+      });
+    });
```

Comment:
I know we use this _pattern_ in a couple of places but do we actually need `next` and then `afterRender`?

Reply:
This is a copy/paste of the old implementation, I'd prefer to leave it as-is since we're deprecating it